main/unbound/unbound.initd

```diff
@@ -22,36 +22,22 @@ depend() {
 	after auth-dns
 }
 
-checkconfig() {
-	checkpath --directory /var/run/unbound
-	UNBOUND_PIDFILE=$("${UNBOUND_CHECKCONF}" -o pidfile "${UNBOUND_CONFFILE}")
-	return $?
-}
-
 configtest() {
-	ebegin "Checking ${SVCNAME} configuration"
-	checkconfig
-	eend $?
-}
-
-start() {
-	checkconfig || return $?
-	ebegin "Starting ${SVCNAME}"
-	start-stop-daemon --start --pidfile "${UNBOUND_PIDFILE}" \
-		--exec "${UNBOUND_BINARY}" -- -c "${UNBOUND_CONFFILE}"
+	ebegin "Checking $RC_SVCNAME configuration"
+	# unbound uses the pid in $cfgfile & we set it via initd
+	sed -i "s|.*pidfile:.*|pidfile:  \"$pidfile\"|" "$cfgfile"
+	unbound-checkconf "$cfgfile" 1>/dev/null
 	eend $?
 }
 
-stop() {
-	checkconfig || return $?
-	ebegin "Stopping ${SVCNAME}"
-	start-stop-daemon --stop --pidfile "${UNBOUND_PIDFILE}"
-	eend $?
+start_pre() {
+	checkpath ${pidfile%/*} 2>/dev/null
```

Comment:
Shouldn't it be `checkpath --directory` here?

I don't like that we hide stderr, how would user otherwise know if something went wrong?

Reply:
whole PR redone - should be all ok now:
```
root@dns [/etc/init.d]# rc-service unbound start
 * Checking: /etc/unbound/unbound.conf
 * Using pidfile: /run/unbound.pid
 * Starting unbound daemon ... [ ok ]
```